src/ivalue_manager.rs

```diff
@@ -46,11 +46,16 @@ fn update<F: FnMut(IValue) -> Result<Option<IValue>, Error>>(
                 let obj = target_once.as_object_mut().unwrap();
                 if is_last {
                     if let Entry::Occupied(mut e) = obj.entry(token) {
-                        let v = e.insert(IValue::NULL);
-                        if let Some(res) = (func)(v)? {
-                            e.insert(res);
-                        } else {
-                            e.remove();
+                        let mut v = e.get_mut();
+                        match (func)(&mut v) {
+                            Ok(res) => {
+                                if let Some(res) = res {
+                                    *v = res;
```

Comment:
Maybe I am missing something but how this API allows me to edit the existing entry and keep it?

Reply:
Lets explain this in a comment in the code and also on top comment on the PR.